nengo/utils/tests/test_socket.py

```diff
@@ -0,0 +1,151 @@
+import math
+import threading
+import numpy as np
+
+import nengo
+from nengo.utils import sockets
+
+
+class SimThread(threading.Thread):
```

Comment:
I'm just generally confused about how this class is supposed to work in general. I can't match it to any of the threading patterns I've seen before. I understand that because of the GIL, the threads will switch eventually after sending a UDP message, since that's an IO process, but what's with all the locking? What shared state is there to protect?

Reply:
is this ensuring that the threads are started in a certain order, or just that they're all ready to start simulating at the same time?